File: params.m
function a = params()

% Geometry
a.dx0 = 0.19; % X distance of first rotor  (m)
a.dx1 = 0.19; % X distance of second rotor (m)
a.dx2 = 0.19; % X distance of third rotor  (m)
a.dx3 = 0.19; % X distance of fourth rotor (m)

a.dy0 = 0.19; % Y distance of first rotor  (m)
a.dy1 = 0.19; % Y distance of second rotor (m)
a.dy2 = 0.19; % Y distance of third rotor  (m)
a.dy3 = 0.19; % Y distance of fourth rotor (m)

% drag
a.cT = 3.75e-9; %0.05; % Rotor drag torque constant (idk)

% Mass parameters
a.mQ = 0.547; % mass of quadrotor (kg)
a.J = [0.0033;0.0033;0.0033];
a.maxThrust = 5; % maximum thrust (N)
a.minThrust = 0; % minimum thrust (N)
a.mL = 0.05; % mass of payload (kg)
a.l = 0.3; % length of cable [m]

% MPC Parameters
a.Q = zeros(3);
a.Q(1:3,1:3) = 40*eye(3);
% a.Q(3,3) = 10;
% a.Q(4:7,4:7) = 0.01*eye(4);
a.R = 0.01*eye(4);
a.predHorizon = 8;
a.dt = 0.05;
a.equalityTolerance = 0;

% Optimizer Parameters
tolerance = 0.1;
a.options = odeset('RelTol',tolerance, 'AbsTol',tolerance);
a.fsolveOptions = optimset('display','none','TolX',1e-6,'TolFun',1e-6,"Algorithm",'sqp','MaxFunEvals',64000,'UseParallel',false);

% Plot?
a.plot = true;

File: params.py
import casadi as ca
import numpy as np

# Geometry
dx0 = 0.19  # X distance of first rotor  (m)
dx1 = 0.19  # X distance of second rotor (m)
dx2 = 0.19  # X distance of third rotor  (m)
dx3 = 0.19  # X distance of fourth rotor (m)

dy0 = 0.19  # Y distance of first rotor  (m)
dy1 = 0.19  # Y distance of second rotor (m)
dy2 = 0.19  # Y distance of third rotor  (m)
dy3 = 0.19  # Y distance of fourth rotor (m)

# drag
cT = 3.75e-9  #0.05  # Rotor drag torque constant (idk)

# Mass parameters
mQ = 0.547  # mass of quadrotor (kg)
J = ca.DM([0.0033, 0.0033, 0.0033])
maxThrust = 5  # maximum thrust (N)
minThrust = 0  # minimum thrust (N)
mL = 0.05  # mass of payload (kg)
l = 1.5  # length of cable [m]

# MPC Parameters
Q = 40*ca.DM_eye(3) 
R = 0.01*ca.DM_eye(4) 
predHorizon = 8 
dt = 0.05 
equalityTolerance = 0 

# Optimizer Parameters
tolerance = 0.1 

# Plot?
plot = True 

initialState = np.array([-0.01, 0, -0.3, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])
# initialStateDCM = np.array([-0.01, 0, -0.3, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0])


# sets initial load position to be l (cable length) away from quadrotor
if np.linalg.norm(initialState[0:3]) != 0:
    initialState[0:3] = l*initialState[0:3] / np.linalg.norm(initialState[0:3])


# q (unit vector between quadrotor and load) initial condition
initialState[13:16] = (initialState[0:3] - [0,0,0]) / l
